packages/ui/src/forms/Form/fieldReducer.ts

```diff
@@ -28,11 +28,14 @@ export function fieldReducer(state: FormState, action: FieldAction): FormState {
 
       const newRow: Row = {
         id: (subFieldState?.id?.value as string) || new ObjectId().toHexString(),
-        blockType: blockType || undefined,
         collapsed: false,
         isLoading: true,
       }
 
+      if (blockType) {
+        newRow.blockType = blockType
```

Comment:
The `undefined` block type will still be processed on the server when building form state. See this line: https://github.com/payloadcms/payload/blob/b9832f40e41fff0a5a5b7bf7b08032aa2d6e8b4e/packages/ui/src/forms/fieldSchemasToFormState/addFieldStatePromise.ts#L413

It's also worth noting that although this strips the `blockType` property out of the array rows, this error is thrown when traversing _blocks_.